src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -130,10 +130,11 @@ void AsyncOpenTask::attach_to_subscription_initializer(AsyncOpenCallback&& async
     // 2. we are instructed to run the subscription initializer via sync_config->rerun_init_subscription_on_open.
     //    But we do that only if this is the first time we open realm.
 
-    auto shared_realm = coordinator->get_realm();
+    auto shared_realm = coordinator->get_realm(nullptr, m_db_first_open);
     const auto init_subscription = shared_realm->get_latest_subscription_set();
+    const auto sub_state = init_subscription.state();
 
-    if (init_subscription.version() == 1 || (rerun_on_launch && m_db_first_open)) {
+    if ((sub_state != sync::SubscriptionSet::State::Complete) || (m_db_first_open && rerun_on_launch)) {
```

Comment:
First part of the check changes the old behavior (with async open taking longer than before because we wait if there is any non-complete query), but I don't see an obvious fix. The comment is not accurate anymore.